### PR TITLE
audit cron fixes for error handling and the global Run

### PR DESCRIPTION
**Pull request by navillasa**
Continues https://github.com/storj/storj/pull/489.

### CODE REVIEW

File: pkg/audit/service.go [hunk 1]
```diff
@@ -73,38 +74,31 @@ func (service *Service) Run(ctx context.Context, interval time.Duration) (err er
...
+			if err != nil {
+				return err
 			}
+		case <-ctx.Done():
+			return ctx.Err()
 		}
-	}()
-
-	return utils.CombineErrors(service.errs...)
+		return nil
```
[the remark this answers is not in the document]
Comment (navillasa) [reply pkg/audit/service.go]: No I'm not sure.

File: pkg/audit/service.go [hunk 2]
```diff
@@ -13,52 +13,62 @@ import (
...
+	APIKey           string        `help:"APIKey to access the statdb"`
+	SatelliteAddr    string        `help:"address to contact services on the satellite"`
+	MaxRetriesStatDB int           `help:"max number of times to attempt updating a statdb batch" default:"3"`
+	Interval         time.Duration `help:"how frequently segments are audited" default:"30s"`
 }
 
 // Run runs the repairer with the configured values
 func (c Config) Run(ctx context.Context, server *provider.Provider) (err error) {
-	service, err := NewService(ctx, c.StatDBPort, c.MaxRetriesStatDB, c.Pointers, c.Transport, c.Overlay, c.ID)
+	ca, err := provider.NewCA(ctx, 12, 4)
```
Comment (egonelbre) [inline pkg/audit/service.go]: I don't think we should be creating identity here, but I think it's fine for now.

File: cmd/satellite/main.go [hunk 3]
```diff
@@ -79,6 +81,7 @@ func cmdRun(cmd *cobra.Command, args []string) (err error) {
 		runCfg.PointerDB,
 		o,
 		runCfg.StatDB,
+		// runCfg.Audit,
```
Comment (jtolds) [inline cmd/satellite/main.go]: how come?

File: pkg/audit/verifier.go [hunk 4]
```diff
@@ -159,20 +161,14 @@ func (d *defaultDownloader) DownloadShares(ctx context.Context, pointer *pb.Poin
...
-		// as in the original slice).
+	copies = make([]infectious.Share, 0, len(originals))
+	for _, original := range originals {
 		if original.Error != nil {
 			continue
 		}
-		copies[i].Data = append([]byte(nil), original.Data...)
-		copies[i].Number = original.PieceNumber
+		copies = append(copies, infectious.Share{
+			Data:   append([]byte(nil), original.Data...),
```
Comment (egonelbre) [inline pkg/audit/verifier.go]: nit, `[]byte{}`, is a bit shorter.

File: pkg/audit/verifier.go [hunk 5]
```diff
@@ -105,6 +106,7 @@ func (d *defaultDownloader) getShare(ctx context.Context, stripeIndex, shareSize
 	if err != nil {
 		return s, err
 	}
+	defer utils.LogClose(rc)
```
Comment (egonelbre) [inline pkg/audit/verifier.go]: Return this error rather than logging it.

**Comment by egonelbre** [comment]
Note, you can merge this without starting it. And do the bug-fixing in a separate PR if you want.

File: pkg/audit/verifier.go (hunk 5 above)
Comment (navillasa) [reply pkg/audit/verifier.go]: I'm using LogClose to close the read closer, so how would I return the error instead of logging?

**Comment by navillasa** [comment]
> Note, you can merge this without starting it. And do the bug-fixing in a separate PR if you want.

I'm not sure how to do that since travis is failing.

**Pull request merged**